tensordict/_td.py

```diff
@@ -1510,21 +1518,26 @@ def save_metadata(data: TensorDictBase, filepath, metadata=None):
                     # ensure subdirectory exists
                     os.makedirs(prefix / key, exist_ok=True)
                     self._tensordict[key] = value.memmap_(
-                        prefix=prefix / key, copy_existing=copy_existing
+                        prefix=prefix / key, copy_existing=copy_existing, executor=executor, num_threads=num_threads,
                     )
                 else:
-                    self._tensordict[key] = value.memmap_()
+                    self._tensordict[key] = value.memmap_(executor=executor, num_threads=num_threads)
                 continue
             else:
                 # user did specify location and memmap is in wrong place, so we copy
-                self._tensordict[key] = MemoryMappedTensor.from_tensor(
-                    value,
-                    filename=str(prefix / f"{key}.memmap")
-                    if prefix is not None
-                    else None,
-                    copy_existing=copy_existing,
-                    # copy_existing=copy_existing,
-                )
+                def _populate():
+                    self._tensordict[key] = MemoryMappedTensor.from_tensor(
+                        value,
+                        filename=str(prefix / f"{key}.memmap")
+                        if prefix is not None
+                        else None,
+                        copy_existing=copy_existing,
+                        # copy_existing=copy_existing,
+                    )
+                if executor is None:
+                    _populate()
+                else:
+                    executor.submit(_populate)
```

Comment:
Probably want to return a Future here so the caller can block on the saves actually completing if they want (or get a callback when it completes)